src/kde/kde.cc

```diff
@@ -0,0 +1,154 @@
+//--------------------------------------------*-C++-*---------------------------------------------//
+/*!
+ * \file   kde/kde.cc
+ * \author Mathew Cleveland
+ * \date   November 10th 2020
+ * \brief  Explicitly defined KDE functions for various dimensions and coordinate
+ *         KDE or Kernel Density Estimators are unbiased statical based
+ *         reconstruction.  They can significantly increase the convergence
+ *         rate of statical distributions. The KDE performs a reconstruction by
+ *         evaluating a mean over some discrete kernel shape. In this DRACO
+ *         implementation the mean is evaluated based on the sample locations
+ *         that are bound by the kernel shape.  A renormalization is used to
+ *         ensure the proper mean is returned given there is no guarantee the
+ *         full kernel (which integrates exactly to 1) will be integrated fully
+ *         in space. This renormalization also avoids the need for boundary
+ *         fix-ups which are typically used in KDE applications to account for
+ *         the kernel extending beyond the bounds of the spatial domain. Other
+ *         approaches that could be considered are quadrature based approaches
+ *         that fully sample the Kernel space reducing the need for the
+ *         normalization.
+ * systems
+ * \note   Copyright (C) 2018-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//------------------------------------------------------------------------------------------------//
+
+#include "kde.hh"
+#include <iostream>
+#include <math.h>
+#include <numeric>
+
+namespace kde {
+
+/*!
+ * reconstruction 
+ * \brief
+ *
+ * Cartesian geometry KDE reconstruction of a 1D distribution
+ *
+ * \param[in] distribution 
+ * \param[in] position
+ * \param[in] band_width
+ * \param[inout] result returned final local function distribution
+ *
+ */
+template <>
+template <>
```

Comment:
Is your plan to have this cartesian version fully templated on 'dim' at some point, such that position and bandwidth are vectors of arrays of size 'dim'? Or will you leave the arrays size 3 regardless of dim? (I think our code always sizes the position vector to 3, so maybe this is just the easier way to go...)

Reply:
Yap, this is why I chose to leave them as a vector 3. Also to make the interface as consistent as possible to between implementations.

Reply:
The templating does allow me to only call global reductions on the dimensional variables that I use though.

Reply:
Cool beans, thanks :)